sidechain/consensus/aura/src/block_importer.rs

```diff
@@ -159,6 +179,61 @@ impl<
 	pub(crate) fn block_author_is_self(&self, block_author: &SignedSidechainBlock::Public) -> bool {
 		self.authority.public() == *block_author
 	}
+
+	fn create_finish_game_extrinsic(
+		&self,
+		sidechain_block: &&<SignedSidechainBlock as SignedBlock>::Block,
+	) -> Result<(), ConsensusError> {
+		let calls = self
+			.top_pool_executor
+			.get_trusted_calls(&sidechain_block.shard_id())
+			.map_err(|e| ConsensusError::Other(format!("{:?}", e).into()))?;
+		for call in calls {
+			if let TrustedCall::connectfour_play_turn(account, _b) = call.call {
+				let shard = &sidechain_block.shard_id();
```

Comment:
You're initializing shard here for every call, even though it will stay the same. Why not at the beginning of the function?